### PR TITLE
Minor usability improvements prior to adding documentation 

### PR DESCRIPTION
**Pull request by x3ro**
Set-up for #2724, making things a little easier to use and spot when something goes wrong.

### CODE REVIEW

File: Makefile.include [hunk 1]
```diff
@@ -287,6 +287,9 @@ term: $(filter flash, $(MAKECMDGOALS))
 		exit 1; }
 	$(TERMPROG) $(TERMFLAGS)
 
+list-ttys:
+	$(RIOTBASE)/dist/tools/usb-serial/list-ttys.sh
```
Comment (daniel-k) [inline Makefile.include]: Please prepend `$(AD)` to suppress output of the command:
`$(AD)$(RIOTBASE)/dist/tools/usb-serial/list-ttys.sh`

File: boards/samr21-xpro/Makefile.include [hunk 2]
```diff
@@ -16,8 +16,12 @@ include $(RIOTBOARD)/Makefile.include.serial
 # Use /dist/tools/usb-serial/list-ttys.sh to find out serial number.
 #   Usage: SERIAL="ATML..." BOARD="samr21-xpro" make flash
 ifneq (,$(SERIAL))
-export OPENOCD_EXTRA_INIT += "-c cmsis_dap_serial $(SERIAL)"
-PORT_LINUX := $(shell $(RIOTBASE)/dist/tools/usb-serial/find-tty.sh $(SERIAL))
+    export OPENOCD_EXTRA_INIT += "-c cmsis_dap_serial $(SERIAL)"
+    SERIAL_TTY=$(shell $(RIOTBASE)/dist/tools/usb-serial/find-tty.sh $(SERIAL))
```
Comment (daniel-k) [inline boards/samr21-xpro/Makefile.include]: Adding some whitespace around the `=` would be nice, but that's just to satisfy my whitespace obsession :smile:

**Comment by daniel-k** [comment]
I like that! Finding that tool was always not that obvious and aborting if a wrong `SERIAL` is specified is actually a really good idea. :+1:

File: Makefile.include (hunk 1 above)
Comment (x3ro) [inline Makefile.include]: It doesn't print anything, then. It's supposed to print something :D

File: boards/samr21-xpro/Makefile.include [hunk 3]
```diff
@@ -16,8 +16,12 @@ include $(RIOTBOARD)/Makefile.include.serial
 # Use /dist/tools/usb-serial/list-ttys.sh to find out serial number.
 #   Usage: SERIAL="ATML..." BOARD="samr21-xpro" make flash
 ifneq (,$(SERIAL))
-export OPENOCD_EXTRA_INIT += "-c cmsis_dap_serial $(SERIAL)"
-PORT_LINUX := $(shell $(RIOTBASE)/dist/tools/usb-serial/find-tty.sh $(SERIAL))
+    export OPENOCD_EXTRA_INIT += "-c cmsis_dap_serial $(SERIAL)"
+    SERIAL_TTY = $(shell $(RIOTBASE)/dist/tools/usb-serial/find-tty.sh $(SERIAL))
+    ifeq (,$(SERIAL_TTY))
+        $(error Did not find a device with serial $(SERIAL))
```
Comment (BytesGalore) [inline boards/samr21-xpro/Makefile.include]: hmm, doesn't seem to fire when no device is connected, else it works :+1:

Comment (daniel-k) [inline boards/samr21-xpro/Makefile.include]: Sure, this will only be executed if `SERIAL` is set. Otherwise it'll pick the platform default.

**Comment by daniel-k** [comment]
ACK :+1:

**Comment by PeterKietzmann** [comment]
and go

**Pull request merged**